HabitTracker.Rutwik2005/HabitTracker.Rutwik2005/Program.cs

```diff
@@ -0,0 +1,230 @@
+                    break;
+                case "2":
+                    Insert();
+                    break;
+                case "3":
+                    Delete();
+                    break;
+                case "4":
+                    Update();
+                    break;
+                default:
+                    Console.WriteLine("\nInvalid Command. Please type a number from 0 to 4.\n");
+                    break;
+            }
+        }
+    }
+    private static void Insert()
+    {
+        string date = GetDateInput();  // Get the date input
+        int quantity = GetNumberInput("\n\nPlease insert number of glasses or other measure of your choice, no decimals allowed\n\n");
+
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+
+            // Use a parameterized query to prevent SQL injection
+            tableCmd.CommandText = "INSERT INTO drinking_water(date, quantity) VALUES(@date, @quantity)";
+
+            tableCmd.Parameters.AddWithValue("@date", date);
+            tableCmd.Parameters.AddWithValue("@quantity", quantity);
+
+            tableCmd.ExecuteNonQuery();
+            connection.Close();
+        }
+
+        Console.WriteLine("\nRecord inserted successfully!\n");
+    }
+
+    internal static string GetDateInput()
+    {
+        Console.WriteLine("\n\nPlease insert the date: (Format: dd-mm-yy). Type 0 to return to main manu.\n\n");
+
+        string dateInput = Console.ReadLine();
+
+        if (dateInput == "0") GetUserInput();
+
+        while (!DateTime.TryParseExact(dateInput, "dd-MM-yy", new CultureInfo("en-US"), DateTimeStyles.None, out _))
+        {
+            Console.WriteLine("\n\nInvalid date. (Format: dd-mm-yy). Type 0 to return to main manu or try again:\n\n");
+            dateInput = Console.ReadLine();
+        }
+
+        return dateInput;
+    }
+    internal static int GetNumberInput(string message)
+    {
+        Console.WriteLine(message);
+        string numberInput = Console.ReadLine();
+        if (numberInput == "0")
+        {
+            GetUserInput();
+        }
+        
+        while(!Int32.TryParse(numberInput, out _)|| Convert.ToInt32(numberInput) < 0)
+        {
+            Console.WriteLine("\n\nInvalid number. Try again.\n\n");
+            numberInput = Console.ReadLine();
+        }
+        int finalInput = Convert.ToInt32(numberInput);
+        return finalInput;
+    }
+    private static void GetAllRecords()
+    {
+        Console.Clear();
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+            tableCmd.CommandText =
+                $"SELECT * FROM drinking_water";
+
+            List<DrinkingWater> tableData = new List<DrinkingWater>();
+
+            SqliteDataReader reader = tableCmd.ExecuteReader();
```

Comment:
`SqliteDataReader` objects (`reader`) should be wrapped in a using statement ideally